test/integration/scheduled-test.js

```diff
@@ -0,0 +1,46 @@
+let sinon = require('sinon')
+let test = require('tape')
+let { join } = require('path')
+let { scheduled } = require('../../src')
+let rateSpy = require('../mock/normal/src/scheduled/rate-scheduled')
+let cronSpy = require('../mock/normal/src/scheduled/cron-scheduled')
+
+const timeoutPromise = time => {
+  return new Promise(resolve => setTimeout(resolve, time))
+}
+
+test('Runs scheduled events from configuration', async t => {
+  process.chdir(join(__dirname, '..', 'mock', 'normal'))
+  await scheduled.end()
+
+  this.clock = sinon.useFakeTimers()
+
+  t.plan(3)
+
+  await rateSpy.resetHistory()
+  await cronSpy.resetHistory()
+
+  await scheduled.start({})
+
+  this.clock.tick(240000)
+
+  this.clock.restore()
+
+  // waiting for the async sub processes to finish
+  await timeoutPromise(5000)
```

Comment:
so this is not 💯 and the 'correct' thing would need to have a private function `_flush` that waits for all currently running sub processes to finish
However i feel like that's a pretty big step compared to what we have now, and it would run the risk of introducing a bunch of memory leaks.

Reply:
Np! Fwiw if there's a way to reduce our reliance on Sinon, my goal is to eventually phase it out – I find it massively clunky and burdensome to maintain, and have developed a strong preference for alternate means of mocking and spying. If that's not realistic here though please do lmk!